src/main/java/hudson/plugins/s3/callable/S3GzipCallable.java

```diff
@@ -21,23 +24,47 @@ public S3GzipCallable(String accessKey, Secret secretKey, boolean useRole, Desti
     @Override
     public String invoke(FilePath file) throws IOException, InterruptedException {
         final File localFile = File.createTempFile("s3plugin", ".bin");
+        Upload upload = null;
 
-        try (InputStream inputStream = file.read()) {
-            try (OutputStream outputStream = new FileOutputStream(localFile)) {
-                try (OutputStream gzipStream = new GZIPOutputStream(outputStream, true)) {
-                    IOUtils.copy(inputStream, gzipStream);
-                    gzipStream.flush();
+        try {
+            try (InputStream inputStream = file.read()) {
```

Comment:
My bad :( Can you extract this 3-levels-try-with-resouces to extra method? I.e.
```
 private void gzipData(FilePath file, File localFile) ... {
        try (InputStream inputStream = file.read()) {
            try (OutputStream outputStream = new FileOutputStream(localFile)) {
                try (OutputStream gzipStream = new GZIPOutputStream(outputStream, true)) {
                    IOUtils.copy(inputStream, gzipStream);
                    gzipStream.flush();
```
It should be possible to do.